snapcraft/project/_project_info.py

```diff
@@ -14,16 +14,86 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-from typing import Any, Dict
+import codecs
+import os
+from collections import OrderedDict
+from copy import deepcopy
+
+import yaml
+import yaml.reader
+
+from . import errors
 
 
 class ProjectInfo:
     """Information gained from the snap's snapcraft.yaml file."""
 
-    def __init__(self, data: Dict[str, Any]) -> None:
-        self.name = data['name']
-        self.version = data.get('version')
-        self.summary = data.get('summary')
-        self.description = data.get('description')
-        self.confinement = data.get('confinement')
-        self.grade = data.get('grade')
+    def __init__(self, *, snapcraft_yaml_file_path) -> None:
+        self.snapcraft_yaml_file_path = get_snapcraft_yaml()
```

Comment:
This should probably be using `snapcraft_yaml_file_path`, no?

Reply:
err, yes!